CMakeLists.txt

```diff
@@ -300,6 +301,9 @@ if(HAIKU)
 endif()
 
 if(NOT CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-builtin-macro-redefined -D'__FILE__=\"$(subst $(realpath ${CMAKE_SOURCE_DIR})/,,$(abspath $<))\"'")

```

Comment:
Can you please add a comment explaining what this is for?

Reply:
It changes __FILE__ to only have the project relative path rather than a full system path. 

The full path is a bit long for the error messages, but also it case confuse some users when they see it in the debug messages: https://youtu.be/jLFzrsPp6os?t=608